dbms/src/Flash/tests/gtest_split_tasks.cpp

```diff
@@ -0,0 +1,140 @@
+// Copyright 2022 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <TestUtils/ExecutorTestUtils.h>
+#include <TestUtils/mockExecutor.h>
+
+namespace DB
+{
+namespace tests
+{
+class SplitTaskTest : public DB::tests::ExecutorTest
+{
+public:
+    void initializeContext() override
+    {
+        ExecutorTest::initializeContext();
+        context.addMockTable({"test_db", "test_table_1"}, {{"s1", TiDB::TP::TypeString}, {"s2", TiDB::TP::TypeString}, {"s3", TiDB::TP::TypeString}});
+        context.addMockTable({"test_db", "l_table"},
+                             {{"s", TiDB::TP::TypeString}, {"join_c", TiDB::TP::TypeString}});
+        context.addMockTable({"test_db", "r_table"},
+                             {{"s", TiDB::TP::TypeString}, {"join_c", TiDB::TP::TypeString}});
+    }
+};
+
+TEST_F(SplitTaskTest, aggregation)
+try
+{
+    auto tasks = context.scan("test_db", "test_table_1")
+                     .filter(eq(col("s2"), col("s3")))
+                     .aggregation({Max(col("s1"))}, {col("s2"), col("s3")})
+                     .filter(eq(col("s2"), col("s3")))
+                     .topN("s2", false, 10)
+                     .buildMPPTasks(context);
+
+    size_t task_size = tasks.size();
+    std::vector<String> executors = {
```

Comment:
Looks cool. But what if swap `executors[0]` and `executors[1]` , it looks more comfortable. Because it's just like the output of `explain`!

Reply:
Optional, not required.

Reply:
Rest LGTM